### ISSUE TITLE
CSS Grid

### ISSUE THREAD

**Issue by Immortalin** [linebender/druid#1383]
A Grid implementation would complement Flexbox well for more complex layouts.

**Comment by cmyr** [comment]
I agree that we should have some sort of grid widget; given our layout model I think the best inspiration/model may be flutter's [GridView](https://api.flutter.dev/flutter/widgets/GridView-class.html).

**Comment by kud1ing** [comment]
This goal might overlap with https://github.com/linebender/druid-widget-nursery:

> A really good, general widget for laying out collections of items.
> There are different approaches we could use (flexbox or grid from the HTML world).

**Comment by alexmozaidze** [comment]
When I was writing HTML/CSS, Grid was super useful, it would be really cool to have something similar in Druid!
